data-prepper-plugins/failures-common/src/main/java/org/opensearch/dataprepper/plugins/dlq/DlqPushHandler.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.dlq;
+
+import io.micrometer.core.instrument.Counter;
+import org.opensearch.dataprepper.metrics.MetricNames;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.configuration.PluginSetting;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+import static org.opensearch.dataprepper.logging.DataPrepperMarkers.NOISY;
+import org.opensearch.dataprepper.model.failures.DlqObject;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+
+/**
+ * * An Handler class which helps log failed data to AWS S3 bucket or file based on configuration.
+ */
+
+public class DlqPushHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(DlqPushHandler.class);
+    public static final String NUM_DLQ_SUCCESS = "NumDlqSuccess";
+    public static final String NUM_DLQ_FAILED = "NumDlqFailed";
+    public static final String STS_ROLE_ARN = "sts_role_arn";
+    public static final String REGION = "region";
+
+    private final DlqProvider dlqProvider;
+    private final PluginSetting dlqPluginSetting;
+    private final DlqWriter dlqWriter;
+    private final Counter dlqSuccessCounter;
+    private final Counter dlqFailedCounter;
+
+    public DlqPushHandler(final PluginFactory pluginFactory, final PluginSetting pluginSetting, final PluginMetrics pluginMetrics,
+                          final PluginModel dlqConfig, final String region, final String role, final String metricsPrefix) {
+        Map<String, Object> dlqSettings = dlqConfig.getPluginSettings();
```

Comment:
What about the sts_header_overrides

Reply:
I am just following what we do for opensearch sink dlq. There is no point in doing much different because pipeline DLQ will make replace this.

Reply:
But you aren't doing it exactly the same. In OpenSearch sink we pass the full dlq config here (https://github.com/opensearch-project/data-prepper/blob/f277b5f798f579d5181cd75163988715c47b24a9/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java#L196), which includes sts header overrides

Reply:
@graytaylor0 Yes. That's there in `data-prepper-plugins/failures-common/src/main/java/org/opensearch/dataprepper/plugins/dlq/s3/` The config is under `s3:` not under `dlq:` It is already supported

Reply:
So if users pass

```
dlq:
  s3:
    sts_header_overrides:
        ....
```
to cloudwatch sink those will get applied?

Reply:
Yes